frontend/public/components/hpa.tsx

```diff
@@ -31,8 +34,113 @@ const menuActions = [
   ...common,
 ];
 
-export const HorizontalPodAutoscalersDetails: React.SFC<HorizontalPodAutoscalersDetailsProps> = ({obj: hpa}) => {
-  return <div className="co-m-pane__body">
+const MetricsRow: React.SFC<MetricsRowProps> = ({type, current, target}) => <div className="row">
+  <div className="col-xs-6">
+    {type}
+  </div>
+  <div className="col-xs-3">
+    {current || '-'}
+  </div>
+  <div className="col-xs-3">
+    {target || '-'}
+  </div>
+</div>;
+
+const externalRow = (metric, current, key) => {
+  const { external } = metric;
+  const type = external.metricName;
+
+  let currentValue, targetValue;
+  // TODO: show metric selector for external metrics?
+  if (external.targetAverageValue) {
+    currentValue = _.get(current, 'object.currentAverageValue');
+    targetValue = external.targetAverageValue;
+  } else if (external.targetValue) {
+    currentValue = _.get(current, 'object.currentValue');
+    targetValue = external.targetValue;
+  }
+
+  return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
+};
+
+const objectRow = (metric, current, key, ns) => {
+  const { object } = metric;
+  const type = <React.Fragment>
+    {object.metricName} on
+    <ResourceLink kind={object.target.kind} name={object.target.name} namespace={ns} title={object.target.name} />
+  </React.Fragment>;
+  const currentValue = _.get(current, 'object.currentValue');
+  const targetValue = object.targetValue;
+
+  return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
+};
+
+const podRow = (metric, current, key) => {
+  const { pods } = metric;
+  const type = `${pods.metricName} on pods`;
```

Comment:
This instance of `pods` does not need to be linked (to pod list page) correct? We are just linking to specific resources, like [line 70](https://github.com/openshift/console/pull/97/files#diff-c721d4405104cb163ef8cda009eb4bf1R70).

Reply:
Yeah we can't link directly to the pods because we don't know what label selector to use to filter the list. It's whatever pods are owned by the thing being scaled.

Reply:
Ok great.